grpclog/component.go

```diff
@@ -49,21 +49,20 @@ var environmentVars = map[string]*componentData{}
 var prefixVars = map[string]*componentData{}
 var cache = map[string]*componentData{}
 
-// init creates all grpc components and any components named in the environment
-// variable and applies all settings specified in the environment variable.
-// Notably, settings for all components with a given prefix are applies before
-// regular component settings.
+// init extracts the component settings from the environment variable.
 func init() {
 	// Pull environment variable data and put in environmentVars and prefixVars
-	if envVar, ok := os.LookupEnv(envName); ok && len(envVar) > 0 {
-		environmentVars, prefixVars = parseEnvironmentVar(envVar)
-	}
+	v, _ := os.LookupEnv(envName)
+	environmentVars, prefixVars = parseEnvironmentVar(v)
 }
 
 // parseEnvironmentVar parses an environment variable string and pulls the component settings data.
 func parseEnvironmentVar(envVar string) (map[string]*componentData, map[string]*componentData) {
 	envVars := map[string]*componentData{}
 	preVars := map[string]*componentData{}
+	if len(envVar) == 0 {
+		return envVars, preVars
+	}
```

Comment:
This is technically not needed, since `strings.Split("", ",")` returns `[]`.

Reply:
I added an test for the empty string, and for some reason it triggers one of the error fmt.Printlns inside the loop without that if condition.

Reply:
Oh, it's not `[]`, it's `[""]` - but that prints as `[]` with `fmt.Print`.

OK.  It's usually better to verify your inputs instead of relying on values to be checked before being called anyway, so this way is better than the previous version IMO.